uploading an app (#2082)
saucelabs/sauce-docs: docs/test-fairy/using-test-fairy/uploading-an-app.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Uploading an Application

> Once your account is created and verified, you are ready to upload your app.

Reviewer: ```suggestion
Once your account is created and verified, you can upload your app.
```

> To upload an app, in the top menu, click **New Upload**.

Reviewer: ```suggestion
Click **New Upload** to upload an app.
```

We recommend using the Upload API method to enable the Jenkins plugin, Gradle plugin, or Command line uploader.

The code of our command line uploader, Jenkins plugin, and Gradle plugin is open source, so feel free to change and improve it.

## Manual Upload

> ### Supported Plaforms

Reviewer: ```suggestion
### Supported Platforms
```

> - **Android**: TestFairy supports uploading and distributing Android Applications. In order to distribute Android apps with TestFairy, they must be packaged as an `.apk` or `.aab` file.

Reviewer: ```suggestion
- **Android**: TestFairy supports uploading and distributing Android Applications. Distributing Android apps with TestFairy must be packaged as a `.apk` or `.aab` file.
```

> - **iOS**: TestFairy supports uploading and distributing iOS applications. iOS apps can be signed with AdHoc, Development, or Enterprise certificates. To distribute iOS apps with TestFairy, they must be packaged as an `.ipa` file.

Reviewer: ```suggestion
- **iOS**: TestFairy supports uploading and distributing iOS applications. You can sign iOS apps with AdHoc, Development, or Enterprise certificates. Distributing iOS apps with TestFairy must be packaged as a `.ipa` file.
```

- **MacOS**: MacOS apps are bundled as `.app` files, however, to distribute MacOS apps with TestFairy, those `.app` files must be zipped into a `.zip` file.

### Choose your Build File

> In the first stage you need to choose the file you want to upload. It can be an **.ipa** (for iOS), **.apk** (for Android), or **.zip** (for MacOS) file.

Reviewer: ```suggestion
In the first stage, choose the file you want to upload. It can be a `.ipa` (for iOS), `.apk` (for Android), or `.zip` (for MacOS) file.
```

### Project Settings

You can define your build settings during the upload process:

> -- **In-app reporting** - Check this box to enable/disable the "shake to report" feature in your app. When enabled, users can shake their device to send a feedback report, along with a video recording, screenshots, logs, and metrics of their test.

Reviewer: ```suggestion
-- **In-app reporting** - Check this box to enable/disable your app's "shake to report" feature. When enabled, users can shake their device to send a feedback report, video recording, screenshots, logs, and test metrics.
```

> -- **Auto-Update** - When auto update is enabled, users using previous versions of this app will get a notification about the new version next time they open up the app. The new version will be downloaded automatically, so the user doesn't have to manually download it. Please note that in this case no email notification will be sent to the testers.

Reviewer: ```suggestion
-- **Auto-Update** - When auto-update is enabled, users using previous versions of this app will get a notification about the new version the next time they open up the app. The latest version will be downloaded automatically, so the user doesn't have to download it manually. Note that no email notification will be sent to the testers in this case.
```

> -- **Custom Comments** - Use this section to add release notes, describe the updates/changes, and anything else you would like your testers to be aware of.

Reviewer: ```suggestion
-- **Custom Comments** - Use this section to add release notes, describe the updates/changes, and anything else you want your testers to know.
```

> **Note:** These settings are relevant only if you added the SDK to your app.

Reviewer: ```suggestion
**Note:** These settings are relevant only if you add the SDK to your app.
```

More build settings are available on the Account Settings page.

### Updating an App

> To update a build, upload the same file again (i.e., the same build with the same version). The new build file will override the old build without creating a new app version.

Reviewer: ```suggestion
To update a build, upload the same file again (that is., the same build with the same version). The new build file will override the old build without creating a new app version.
```

### Uploading a New Version

> To upload a new build (a new version of the same app), upload the new version in the exact same way you uploaded the old version. TestFairy will identify that both apps have the same package name (bundle identifier) and group them together in the same project.

Reviewer: ```suggestion
To upload a new build (a new version of the same app), upload the new version as you uploaded the old version. TestFairy identifies that both apps have the same package name (bundle identifier) and group them together in the same project.
```
